test/runtests.jl

```diff
@@ -2,6 +2,52 @@ using Zygote, Test
 using Zygote: gradient
 using CUDA: has_cuda
 
+using Zygote
+@testset "gradient checkpointing" begin
+    mutable struct CountCalls
+        ncalls
+    end
+    CountCalls() = CountCalls(0)
+    function (o::CountCalls)(x)
+        o.ncalls += 1
+        x
+    end
+
+    f = CountCalls()
+    g = CountCalls()
+    h = CountCalls()
+
+    y, pb = Zygote.pullback(h ∘ g ∘ f, 4.0)
+    @test y === 4.0
+    @test f.ncalls === g.ncalls === h.ncalls === 1
+    @test pb(1.0) === (1.0,)
+    @test f.ncalls === g.ncalls === h.ncalls === 1
+
+    f = CountCalls()
+    g = CountCalls()
+    h = CountCalls()
+    y,pb = Zygote.pullback(Zygote.checkpointed, h∘g∘f, 4.0)
+    @test f.ncalls === g.ncalls === h.ncalls === 1
+    @test pb(1.0) === (1.0,)
+    @test f.ncalls === g.ncalls === h.ncalls === 2
+    @test pb(1.0) === (1.0,)
+    @test f.ncalls === g.ncalls === h.ncalls === 3
+
+    f = CountCalls()
+    g = CountCalls()
+    h = CountCalls()
+    function doit(x)
+        x2 = f(x)
+        Zygote.checkpointed(h∘g, x2)
+    end
+    y,pb = Zygote.pullback(doit, 4.0)
+    @test f.ncalls === g.ncalls === h.ncalls === 1
+    @show pb(1.0)
```

Comment:
This call fails:
```
  Got exception outside of a @test
  BoundsError: attempt to access (nothing, 1.0)
    at index [3]
  Stacktrace:
   [1] getindex at ./tuple.jl:24 [inlined]
   [2] gradindex(::Tuple{Nothing,Float64}, ::Int64) at /home/jan/.julia/dev/Zygote/src/compiler/re
verse.jl:12
   [3] doit at /home/jan/.julia/dev/Zygote/test/runtests.jl:41 [inlined]
   [4] (::typeof(∂(λ)))(::Float64) at /home/jan/.julia/dev/Zygote/src/compiler/interface2.jl:0
   [5] (::Zygote.var"#41#42"{typeof(∂(λ))})(::Float64) at /home/jan/.julia/dev/Zygote/src/compiler
/interface.jl:40
   [6] top-level scope at show.jl:641
   [7] top-level scope at /home/jan/.julia/dev/Zygote/test/runtests.jl:45
   [8] top-level scope at /home/jan/julia/usr/share/julia/stdlib/v1.5/Test/src/Test.jl:1115
   [9] top-level scope at /home/jan/.julia/dev/Zygote/test/runtests.jl:7
   [10] include(::String) at ./client.jl:457
   [11] top-level scope at none:6
   [12] eval(::Module, ::Any) at ./boot.jl:331
   [13] exec_options(::Base.JLOptions) at ./client.jl:272
   [14] _start() at ./client.jl:506
```
what am I doing wrong?

Reply:
Ah, sorry, the code I linked in the issue has a bug. Currently, the pullback isn't returning a cotangent for `checkpointed` itself. It should just be nothing, so returning something like
```julia
(nothing, pb(Δy)...)
```
should be sufficient.